code/src/iris/src/lib/services/repository-service.ts

```diff
@@ -0,0 +1,199 @@
+import type { Repository, Installation } from '../types';
+import { api } from '../api';
+import { cachedRepositories, repositoriesCacheLoading } from '../stores';
+
+export interface RepositoryWithStats extends Repository {
+  runCount?: number;
+  lastRun?: string;
+}
+
+export interface RepositoryLoadResult {
+  repositories: RepositoryWithStats[];
+  fromCache: boolean;
+  error?: string;
+}
+
+/**
+ * Centralized repository loading service with persistent caching
+ */
+export class RepositoryService {
+  private static instance: RepositoryService;
+  private loadingPromises: Map<string, Promise<RepositoryLoadResult>> = new Map();
+
+  private constructor() {}
+
+  static getInstance(): RepositoryService {
+    if (!RepositoryService.instance) {
+      RepositoryService.instance = new RepositoryService();
+    }
+    return RepositoryService.instance;
+  }
+
+  /**
+   * Load all repositories for an installation, using cache if available
+   * @param installation The installation to load repositories for
+   * @param forceRefresh If true, bypasses cache and loads fresh data
+   * @returns Promise with repositories and metadata
+   */
+  async loadRepositories(
+    installation: Installation, 
+    forceRefresh: boolean = false
+  ): Promise<RepositoryLoadResult> {
+    const installationId = installation.id;
+
+    // If already loading for this installation, return the existing promise
+    const existingPromise = this.loadingPromises.get(installationId);
+    if (existingPromise && !forceRefresh) {
+      return existingPromise;
+    }
+
+    // Create new loading promise
+    const loadPromise = this._loadRepositoriesInternal(installation, forceRefresh);
+    this.loadingPromises.set(installationId, loadPromise);
+
+    try {
+      const result = await loadPromise;
+      return result;
+    } finally {
+      // Clean up loading promise
+      this.loadingPromises.delete(installationId);
+    }
+  }
+
+  private async _loadRepositoriesInternal(
+    installation: Installation,
+    forceRefresh: boolean
```

Comment:
What does "force refresh" actually do?  It doesn't seem to call the endpoint to force a refresh of the repos, is it just about not using cached repos?